packages/swapper/src/utils.ts

```diff
@@ -169,16 +170,25 @@ export const checkEvmSwapStatus = async ({
   txHash,
   chainId,
   assertGetEvmChainAdapter,
+  getSafeApiKit,
 }: {
   txHash: string
   chainId: ChainId
   assertGetEvmChainAdapter: (chainId: ChainId) => EvmChainAdapter
+  getSafeApiKit: (chainId: EvmChainId, safeAddress: string) => Promise<SafeApiKit>
 }): Promise<{
   status: TxStatus
   buyTxHash: string | undefined
   message: string | undefined
 }> => {
   try {
+    const safeApiKit = await getSafeApiKit(
+      chainId as EvmChainId,
+      '0x00342Dc97c6419089130bB2F1a39c1Bfea4fFe40',
+    )
+    // @ts-ignore
+    const transaction = await safeApiKit.getTransaction(txHash)
+    debugger
```

Comment:
Here she is - that's all we need and will give us something like this:

What matters here:

- `confirmationsRequired` - i.e less than that and we have no actual Txid yet
- `confirmations` - the actual confirmations from signers
- `safeTxHash`, we already have this as `txHash` here - we can use this to generate a [valid Tx link](https://app.safe.global/transactions/tx?id=multisig_0x00342Dc97c6419089130bB2F1a39c1Bfea4fFe40_0xa75d5c9b8fd53cc74a0e35fba841ca1c6f5b8f2ee665a0ac8cf8f93e631a1200&safe=avax:0x00342Dc97c6419089130bB2F1a39c1Bfea4fFe40) while not all signers have signed yet
- `transactionHash` - the actual on-chain hash of the swap. Once we have this, we can use in-place of the SAFE Tx hash and do good ol' status detection

```json
{
    "safe": "0x00342Dc97c6419089130bB2F1a39c1Bfea4fFe40",
    "to": "0xDef1C0ded9bec7F1a1670819833240f027b25EfF",
    "value": "4670714619336759",
    "data": "0x415565b0000000000000000000000000eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee000000000000000000000000b97ef9ef8734c71904d8002f8b6bc66dd9c48a6e000000000000000000000000000000000000000000000000001097fd7dbb983700000000000000000000000000000000000000000000000000000000000184b200000000000000000000000000000000000000000000000000000000000000a0000000000000000000000000000000000000000000000000000000000000000500000000000000000000000000000000000000000000000000000000000000a0000000000000000000000000000000000000000000000000000000000000014000000000000000000000000000000000000000000000000000000000000004e000000000000000000000000000000000000000000000000000000000000005e000000000000000000000000000000000000000000000000000000000000006a0000000000000000000000000000000000000000000000000000000000000000400000000000000000000000000000000000000000000000000000000000000400000000000000000000000000000000000000000000000000000000000000040000000000000000000000000eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee000000000000000000000000000000000000000000000000001097fd7dbb983700000000000000000000000000000000000000000000000000000000000000150000000000000000000000000000000000000000000000000000000000000040000000000000000000000000000000000000000000000000000000000000034000000000000000000000000000000000000000000000000000000000000000200000000000000000000000000000000000000000000000000000000000000000000000000000000000000000b31f66aa3c1e785363f0875a1b74e27b85fd66c7000000000000000000000000b97ef9ef8734c71904d8002f8b6bc66dd9c48a6e00000000000000000000000000000000000000000000000000000000000001400000000000000000000000000000000000000000000000000000000000000300000000000000000000000000000000000000000000000000000000000000030000000000000000000000000000000000000000000000000000000000000002c0000000000000000000000000000000000000000000000000001097fd7dbb98370000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000030000000000000000000000000000000000000000000000000000000000000000010000000000000000000000000000000000000000000000000000000000000020000000000000000000000000000000025472616465724a6f6500000000000000000000000000000000000000000000000000000000000000001097fd7dbb98370000000000000000000000000000000000000000000000000000000000018547000000000000000000000000000000000000000000000000000000000000008000000000000000000000000000000000000000000000000000000000000000a000000000000000000000000060ae616a2155ee3d9a68541ba4544862310933d400000000000000000000000000000000000000000000000000000000000000400000000000000000000000000000000000000000000000000000000000000002000000000000000000000000b31f66aa3c1e785363f0875a1b74e27b85fd66c7000000000000000000000000b97ef9ef8734c71904d8002f8b6bc66dd9c48a6e000000000000000000000000000000000000000000000000000000000000000100000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000b000000000000000000000000000000000000000000000000000000000000004000000000000000000000000000000000000000000000000000000000000000a000000000000000000000000000000000000000000000000000000000000000200000000000000000000000000000000000000000000000000000000000000001000000000000000000000000b97ef9ef8734c71904d8002f8b6bc66dd9c48a6e0000000000000000000000000000000000000000000000000000000000000095000000000000000000000000ad01c20d5886137e056775af56915de824c8fce5000000000000000000000000000000000000000000000000000000000000000d00000000000000000000000000000000000000000000000000000000000000400000000000000000000000000000000000000000000000000000000000000060000000000000000000000000b97ef9ef8734c71904d8002f8b6bc66dd9c48a6e000000000000000000000000000000000000000000000000000000000001c50d00000000000000000000000074d63f31c2335b5b3ba7ad2812357672b2624ced000000000000000000000000000000000000000000000000000000000000000c000000000000000000000000000000000000000000000000000000000000004000000000000000000000000000000000000000000000000000000000000000e00000000000000000000000000000000000000000000000000000000000000020000000000000000000000000000000000000000000000000000000000000004000000000000000000000000000000000000000000000000000000000000000a00000000000000000000000000000000000000000000000000000000000000002000000000000000000000000b31f66aa3c1e785363f0875a1b74e27b85fd66c7000000000000000000000000eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee0000000000000000000000000000000000000000000000000000000000000000869584cd000000000000000000000000c770eefad204b5180df6a14ee197d99d808ee52d0000000000000000000000000000000000000000c074000f48d7f023fed86a23",
    "operation": 0,
    "gasToken": "0x0000000000000000000000000000000000000000",
    "safeTxGas": 0,
    "baseGas": 0,
    "gasPrice": "0",
    "refundReceiver": "0x0000000000000000000000000000000000000000",
    "nonce": 14,
    "executionDate": "2024-08-07T10:39:39Z",
    "submissionDate": "2024-08-07T10:36:46.629789Z",
    "modified": "2024-08-07T10:39:39Z",
    "blockNumber": 48957807,
    "transactionHash": "0x9d6555a68425dcbb21b4d87b5ffa9a2ad9a590fd444ffe228fa1e9fe60bf1eff",
    "safeTxHash": "0xa75d5c9b8fd53cc74a0e35fba841ca1c6f5b8f2ee665a0ac8cf8f93e631a1200",
    "proposer": "0x5daF465a9cCf64DEB146eEaE9E7Bd40d6761c986",
    "executor": "0xb8FF589DbbEc0DDe6401DC9572F54f8EC779b364",
    "isExecuted": true,
    "isSuccessful": true,
    "ethGasPrice": "25000000000",
    "maxFeePerGas": "50000000000",
    "maxPriorityFeePerGas": null,
    "gasUsed": 373645,
    "fee": "9341125000000000",
    "origin": "{\"url\": \"http://localhost:3000\", \"name\": \"ShapeShift DAO\"}",
    "dataDecoded": null,
    "confirmationsRequired": 2,
    "confirmations": [
        {
            "owner": "0x5daF465a9cCf64DEB146eEaE9E7Bd40d6761c986",
            "submissionDate": "2024-08-07T10:36:46.706478Z",
            "transactionHash": null,
            "signature": "0xc66b215fdf0fbb0fe66f978ebcc36df97d97254a8dfa9448157e67e75117a7b5524971609da8446b12e29029e5f320ba4eda38cdec056631ad2fc4cd1d26139c1b",
            "signatureType": "EOA"
        },
        {
            "owner": "0xb8FF589DbbEc0DDe6401DC9572F54f8EC779b364",
            "submissionDate": "2024-08-07T10:39:39Z",
            "transactionHash": null,
            "signature": "0x000000000000000000000000b8ff589dbbec0dde6401dc9572f54f8ec779b364000000000000000000000000000000000000000000000000000000000000000001",
            "signatureType": "APPROVED_HASH"
        }
    ],
    "trusted": true,
    "signatures": "0xc66b215fdf0fbb0fe66f978ebcc36df97d97254a8dfa9448157e67e75117a7b5524971609da8446b12e29029e5f320ba4eda38cdec056631ad2fc4cd1d26139c1b000000000000000000000000b8ff589dbbec0dde6401dc9572f54f8ec779b364000000000000000000000000000000000000000000000000000000000000000001"
}
```